document-store/src/main/java/org/hypertrace/core/documentstore/postgres/Params.java

```diff
@@ -0,0 +1,151 @@
+package org.hypertrace.core.documentstore.postgres;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Holds the params that need to be set in the PreparedStatement for constructing the final SQL
+ * query
+ */
+public class Params {
+
+    // Map of index to the corresponding param value
+    private final Map<Integer, Integer> integerParams;
```

Comment:
Currently, for the `Filter` interface, we have value as `Object` type - https://github.com/hypertrace/document-store/blob/main/document-store/src/main/java/org/hypertrace/core/documentstore/Filter.java
We don't have knowledge of proper types and with which operator what are valid types.

So, as a thought (for avoiding type conversations w/o proper interface) should we go with just an object map here for all params and convert to its string representation during buildPreparedStatement?
```
public class Params {
private final Map<Integer, Object> objectParams;
}
```